main/src/ca/uwaterloo/flix/language/phase/ClosureConv.scala

```diff
@@ -146,6 +170,41 @@ object ClosureConv {
 
   }
 
+  private def rewriteApplyLocalDef(expr00: Expr, sym0: Symbol.VarSym, freeVars: List[FreeVar]): Expr = {
+    def visit(expr0: Expr): Expr = expr0 match {
+      case Expr.Cst(cst, tpe, loc) => Expr.Cst(cst, tpe, loc)
+      case Expr.Var(sym, tpe, loc) => Expr.Var(sym, tpe, loc)
+      case Expr.Lambda(fparams, exp, tpe, loc) => Expr.Lambda(fparams, visit(exp), tpe, loc)
+      case Expr.Apply(exp, args, tpe, purity, loc) => Expr.Apply(visit(exp), args.map(visit), tpe, purity, loc)
+      case Expr.LambdaClosure(cparams, fparams, freeVars, exp, tpe, loc) => Expr.LambdaClosure(cparams, fparams, freeVars, visit(exp), tpe, loc)
+      case Expr.ApplyAtomic(op, exps, tpe, purity, loc) => Expr.ApplyAtomic(op, exps.map(visit), tpe, purity, loc)
+      case Expr.ApplyClo(exp, exps, tpe, purity, loc) => Expr.ApplyClo(visit(exp), exps.map(visit), tpe, purity, loc)
+      case Expr.ApplyDef(sym, exps, tpe, purity, loc) => Expr.ApplyDef(sym, exps.map(visit), tpe, purity, loc)
+      case Expr.ApplyLocalDef(sym, exps, itpe, tpe, purity, loc) =>
+        val es = exps.map(visit)
+        if (sym == sym0) {
+          val all = freeVars.map(fv => Expr.Var(fv.sym, fv.tpe, loc.asSynthetic)) ++ es
+          Expr.ApplyLocalDef(sym, all, itpe, tpe, purity, loc)
+        } else {
+          Expr.ApplyLocalDef(sym, es, itpe, tpe, purity, loc)
+        }
+      case Expr.IfThenElse(exp1, exp2, exp3, tpe, purity, loc) => Expr.IfThenElse(visit(exp1), visit(exp2), visit(exp3), tpe, purity, loc)
```

Comment:
I think we should add "some space" so its easier to read. Maybe also introduce local variables. 

Otherwise this function is just very crammed and it would be easy to miss a small mistake.